src/main/java/org/betonquest/betonquest/database/MySQL.java

```diff
@@ -144,69 +159,88 @@ private void migration1(final Connection connection) throws SQLException {
     @SuppressFBWarnings("SQL_NONCONSTANT_STRING_PASSED_TO_EXECUTE")
     private void migration2(final Connection connection) throws SQLException {
         try (Statement statement = connection.createStatement()) {
-            statement.executeUpdate("CREATE TABLE " + prefix + "profile "
-                    + "(profileID CHAR(36) PRIMARY KEY NOT NULL)");
-            statement.executeUpdate("INSERT INTO " + prefix + "profile " + "(profileID) SELECT playerID FROM "
-                    + prefix + "player");
+            statement.executeUpdate("CREATE TABLE " + prefix + "profile ("
+                    + "profileID CHAR(36) PRIMARY KEY NOT NULL)");
+            statement.executeUpdate("INSERT INTO " + prefix + "profile "
+                    + "(profileID) SELECT playerID FROM " + prefix + "player");
             statement.executeUpdate("ALTER TABLE " + prefix + "backpack "
-                    + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, " + "ADD FOREIGN KEY (profileID) REFERENCES "
-                    + prefix + "profile (profileID) ON DELETE CASCADE");
+                    + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, "
+                    + "ADD FOREIGN KEY (profileID) REFERENCES " + prefix + "profile (profileID) ON DELETE CASCADE");
             statement.executeUpdate("ALTER TABLE " + prefix + "journal "
                     + "CHANGE COLUMN playerID profileID CHAR(36) NOT NULL, "
-                    + "MODIFY COLUMN pointer VARCHAR(255) NOT NULL, " + "ADD FOREIGN KEY (profileID) REFERENCES "
-                    + prefix + "profile (profileID) ON DELETE CASCADE");
-            statement.executeUpdate("DELETE FROM " + prefix + "objectives WHERE id NOT IN "
+                    + "MODIFY COLUMN pointer VARCHAR(255) NOT NULL, "
+                    + "ADD FOREIGN KEY (profileID) REFERENCES " + prefix + "profile (profileID) ON DELETE CASCADE");
+            statement.executeUpdate("DELETE FROM " + prefix + "objectives "
```

Comment:
This query gives error `java.sql.SQLException: You can't specify target table 'bq_objectives' for update in FROM clause`.

Apparently in mysql you can't modify a table using the same table in a subquery: https://stackoverflow.com/questions/45494/mysql-error-1093-cant-specify-target-table-for-update-in-from-clause

Reply:
Hmm, strange, because for me it worked, may it depend on the MySQL implementation where the statements are running. Thanks a lot, then i need to figure out an alternative.

Reply:
Ok, thank you again, i hope i have now a working version